test/mpi.py

```diff
@@ -260,7 +260,7 @@ def test_set_mpiexec_opts_from_env(self):
 
         print("opts_from_env: %s" % mpi_instance.mpiexec_opts_from_env)
         for env_var in mpi_instance.mpiexec_opts_from_env:
-            self.assertTrue(env_var.startswith(tuple(prefixes)),
+            self.assertTrue(env_var.startswith(tuple(prefixes)) or env_var in mpi_instance.OPTS_FROM_ENV_BASE,
```

Comment:
this is only checking in one direction, maybe also add a couple of checks like:

```python
self.assertTrue('PYTHONPATH' in mpi_instance.mpiexec_opts_from_env)
```

You'll need to make sure `PYTHONPATH` is defined in `os.environ` before running `set_mpiexec_opts_from_env()`; also make sure you restore the environment in `tearDown`